### DISCUSSION TITLE
WIP: Minder web client

### DISCUSSION THREAD

**Discussion by AGMETEOR** [stacklok/minder#2047] [Ideas]
Happy new year Stacklokers! I have been learning about the minder tool, mostly the CLI and server, and as I was doing so, I decided to experiment with creating a web client using NextJS. The designs are not that great but hopefully this can get useful. I also have a lot of things to fix but I'll share some screen shots anyway.

/login
---
<img width="1792" alt="Screenshot 2024-01-01 at 6 40 05 PM" src="https://github.com/stacklok/minder/assets/28563179/7cd96c3d-9450-46dc-b3bb-c4a8d771e649">

/projects , switching between project contexts
---
<img width="1792" alt="Screenshot 2024-01-01 at 6 41 08 PM" src="https://github.com/stacklok/minder/assets/28563179/9b6a1470-3b7b-4c32-adb3-2703908a5748">

/dashboard/<project_id>/providers, enroll different project providers
---
<img width="1792" alt="Screenshot 2024-01-01 at 6 42 48 PM" src="https://github.com/stacklok/minder/assets/28563179/fb75e75d-a5e1-4b7c-8270-d45111c9abcb">

/dashboard/<project_id>/rules, create and edit rules from within the dashboard
---
<img width="1792" alt="Screenshot 2024-01-01 at 6 44 52 PM" src="https://github.com/stacklok/minder/assets/28563179/eab49d3f-c605-40ac-97fa-b44a33ba931c">

/dashboard/<project_id>/profiles, create and edit profiles from within the dashboard
---
<img width="1792" alt="Screenshot 2024-01-01 at 6 47 16 PM" src="https://github.com/stacklok/minder/assets/28563179/73a3a14c-aaec-42b0-a8db-8e8f1806fbd0">

@JAORMX @lukehinds @rdimitrov

**Comment by lukehinds** [comment]
This is awesome work @AGMETEOR !

What are your plans for the UI? To be transparent we are developing our own UI internally, but we support community folks in contributing what maybe of value to others and love seeing people build around minder, so we would be happy to link to your repo as a reference use of the UI and if you like you could give a community demo on our discord (we plan to get these underway shortly)?

**Reply by AGMETEOR** [reply]
Ok I worked on this web client as an alternative to the CLI, I didn’t know there were already efforts to build one internally. In the meantime, I’ll try and maintain this one or get it to a point where it’s as useful as the CLI , until the official minder web client is released.

**Reply by AGMETEOR** [reply]
So the goal is to have an initial release next week of this web client. Also, I think it'll be nice to have a community demo and hopefully the community can help me make it better and valuable to everyone. How do I register for these demos?
